### ISSUE TITLE
Infinite loop if client call `ListTunnels` with specific name and auto-pagination

### ISSUE THREAD

**Issue by kamontat** [cloudflare/cloudflare-go#1247]
### Confirmation

- [X] My issue isn't already found on the issue tracker.
- [X] I have replicated my issue using the latest version of the library and it is still present.

### cloudflare-go version

0.63.0

### Go environment

```env
GO111MODULE=""
GOARCH="arm64"
GOBIN=""
GOCACHE="<hidden>/Library/Caches/go-build"
GOENV="<hidden>/Library/Application Support/go/env"
GOEXE=""
GOEXPERIMENT=""
GOFLAGS=""
GOHOSTARCH="arm64"
GOHOSTOS="darwin"
GOINSECURE=""
GOMODCACHE="<hidden>/.asdf/installs/golang/1.20.1/packages/pkg/mod"
GONOPROXY=""
GONOSUMDB=""
GOOS="darwin"
GOPATH="<hidden>/.asdf/installs/golang/1.20.1/packages"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="<hidden>/.asdf/installs/golang/1.20.1/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="<hidden>/.asdf/installs/golang/1.20.1/go/pkg/tool/darwin_arm64"
GOVCS=""
GOVERSION="go1.20.1"
GCCGO="gccgo"
AR="ar"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
GOMOD="<hidden>/go.mod"
GOWORK=""
CGO_CFLAGS="-O2 -g"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-O2 -g"
CGO_FFLAGS="-O2 -g"
CGO_LDFLAGS="-O2 -g"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -arch arm64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/8w/3sgs6h1s55773snm6gqgw8780000gn/T/go-build1902134300=/tmp/go-build -gno-record-gcc-switches -fno-common"
```

### Expected output

Return List of Tunnel.

### Actual output

Infinite loop.

### Code demonstrating the issue

var context, cancelFn = context.WithCancel(context.Background())
var tunnels, _, err = c.api.ListTunnels(context, "<UUID>", cloudflare.TunnelListParams{
    Name: "name",
})

### Steps to reproduce

1. Call api.ListTunnels() with below parameter:

```golang
var tunnels, _, err = c.api.ListTunnels(context, "<UUID>", cloudflare.TunnelListParams{
    Name: "name",
})
```

2. above function will never result.

### Debugging result

This happen because if we call `https://api.cloudflare.com/client/v4/accounts/<AccountID>/cfd_tunnel` with `?name=<name>` 

1. the api didn't result `ResultInfo` as picture below:

![CleanShot 2023-03-26 at 12 37 12@2x](https://user-images.githubusercontent.com/14089557/227757537-f5dfce69-1a53-4128-addd-5472b9e824a3.png)

2. On [this line][tunnel#listtunnels]: `ResultInfo` always be zero as image below.

<img width="533" alt="CleanShot 2023-03-26 at 12 44 49@2x" src="https://user-images.githubusercontent.com/14089557/227757699-bc83c6bb-f221-4d6e-9c49-74d99e460295.png">

3. Then `ResultInfo.Next()` [src][resultinfo#next] will increase Page to 1
4. However, on `ResultInfo.Done()` [src][resultinfo#done], Page must more than 1 and more than TotalPages
5. This cause infinite loop

[tunnel#listtunnels]: https://github.com/cloudflare/cloudflare-go/blob/0a2244303cb0851c57a2202262d16543c62e8aaf/tunnel.go#L223-L227
[resultinfo#next]: https://github.com/cloudflare/cloudflare-go/blob/0a2244303cb0851c57a2202262d16543c62e8aaf/pagination.go#L10-L13
[resultinfo#done]: https://github.com/cloudflare/cloudflare-go/blob/0a2244303cb0851c57a2202262d16543c62e8aaf/pagination.go#L4-L6

**Comment by jacobbednarz** [comment]
thanks, this is a known issue at the moment as the backend service doesn't supply the pagination envelope. the internal team will be addressing it. 

you can work around it at the moment by supplying the `per_page` parameter to a value like 1000.

**Comment by kamontat** [comment]
I didn't saw on any opened issues. This will help others understand what happen.

Thank you.

**Comment by jacobbednarz** [comment]
that's fine; it was noticed in another repository and i notified the internal team. i've leave this one here for posterity. once this is fixed, everything will work by itself again and you shouldn't need to upgrade (only remove your workaround).

**Comment by pratheekrebala** [comment]
Thanks @jacobbednarz! This is also affecting https://github.com/cloudflare/terraform-provider-cloudflare/pull/2364. I will use a large `per_page` value for now.

**Comment by jacobbednarz** [comment]
infinite loop is addressed by #1327. the auto pagination will default to a single page until the service team
